wgpu-core/src/snatch.rs

```diff
@@ -59,6 +64,10 @@ impl<T> std::fmt::Debug for Snatchable<T> {
 
 unsafe impl<T> Sync for Snatchable<T> {}
 
+thread_local! {
+    static READ_LOCK_LOCATION: Cell<Option<(&'static Location<'static>, Backtrace)>> = const { Cell::new(None) };
+}
```

Comment:
I don't quite trust rust to optimize this out when it's not used and thread locals can be a valuable resource. -> Can we put this entire thing under `cfg!(debug_assertions)`?

Reply:
A thread local variable is just turned into an LLVM global, it has no trouble optimizing it out when not used (just like for other types of globals, like `static`s and literals)

Reply:
I'm no expert in this, but that doesn't sound at all - the key might be just a regular global, but at least on some platforms (and circumstances) using a thread local means that there's an allocation of a thread local slot in the thread (see e.g. `pthread_key_create` to obtain a key in the first place; I guess Rust might use a single key for several thread locals and then "subkey", but I figure it has to happen at least once). If this was just a regular global variable, how would allocation per thread work if not being directed to a per thread table of sorts.

Anyways, tried a bit in CompilerExplorer and it seems to optimize out fine as long as the thread local is not `pub`, so the compiler knows that it's indeed not in use.
So no point in dying on this hill regardless (: